presto-main/src/main/java/com/facebook/presto/memory/MemoryPoolListener.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.memory;
+
+import java.util.function.Consumer;
+
+public interface MemoryPoolListener
+{
+    /**
+     * Invoked when memory reservation completes successfully.
+     *
+     * @param memoryPool the {@link MemoryPool} where the reservation took place
+     */
+    default void onMemoryReserved(MemoryPool memoryPool) {}
+
+    /**
+     * Creates {@link MemoryPoolListener} implementing {@link #onMemoryReserved(MemoryPool)} only.
+     */
+    static MemoryPoolListener onMemoryReserved(Consumer<? super MemoryPool> action)
```

Comment:
The name collision is not unintentional. Intent is:

* be able to instantiate interface with a lamba / method expr, almost as if it was `@Fluent`
* but still be able to evolve the interface by adding new methods

So the name of this method should include "`onMemoryReserved`" to convey the returned implementation will implement the `onMemoryReserved` method and that method only. Can you propose some name that satisfies this?